src/ripple/app/consensus/RCLValidations.cpp

```diff
@@ -174,97 +280,78 @@ handleNewValidation(Application& app,
-    boost::optional<PublicKey> pubKey = app.validators().getTrustedKey(signer);
-    if (!val->isTrusted() && pubKey)
+    boost::optional<PublicKey> masterKey =
+        app.validators().getTrustedKey(signingKey);
+    if (!val->isTrusted() && masterKey)
         val->setTrusted();
-    RCLValidations& validations  = app.getValidations();
 
-    beast::Journal j = validations.journal();
+    // If not currently trusted, see if signer is currently listed
+    if (!masterKey)
+        masterKey = app.validators().getListedKey(signingKey);
 
-    // Do not process partial validations.
-    if (!val->isFull())
+    bool shouldRelay = false;
+    RCLValidations& validations = app.getValidations();
+    beast::Journal j = validations.adaptor().journal();
+
+    auto dmp = [&](beast::Journal::Stream s, std::string const& msg) {
+            s << "Val for " << hash
+              << (val->isTrusted() ? " trusted/" : " UNtrusted/")
+              << (val->isFull() ? " full" : "partial") << " from "
+              << toBase58(TokenType::TOKEN_NODE_PUBLIC, *masterKey)
+              << "signing key "
+              << toBase58(TokenType::TOKEN_NODE_PUBLIC, signingKey) << " "
+              << msg
+              << " src=" << source;
+        };
+
+    if(!val->isFieldPresent(sfLedgerSequence))
     {
-        const bool current = isCurrent(
-            validations.parms(),
-            app.timeKeeper().closeTime(),
-            val->getSignTime(),
-            val->getSeenTime());
-
-        JLOG(j.debug()) << "Val (partial) for " << hash << " from "
-                         << toBase58(TokenType::TOKEN_NODE_PUBLIC, signer)
-                         << " ignored "
-                         << (val->isTrusted() ? "trusted/" : "UNtrusted/")
-                         << (current ? "current" : "stale");
-
-        // Only forward if current and trusted
-        return current && val->isTrusted();
+        if(j.error())
+            dmp(j.error(), " missing ledger seqeuence field");
+        return false;
     }
 
-    if (!val->isTrusted())
+    // masterKey is seated only if validator is trusted or listed
+    if (masterKey)
     {
-        JLOG(j.trace()) << "Node "
-                        << toBase58(TokenType::TOKEN_NODE_PUBLIC, signer)
-                        << " not in UNL st="
-                        << val->getSignTime().time_since_epoch().count()
-                        << ", hash=" << hash
-                        << ", shash=" << val->getSigningHash()
-                        << " src=" << source;
-    }
+        ValStatus const outcome = validations.add(*masterKey, val);
 
-    // If not currently trusted, see if signer is currently listed
-    if (!pubKey)
-        pubKey = app.validators().getListedKey(signer);
 
-    bool shouldRelay = false;
 
-    // only add trusted or listed
-    if (pubKey)
-    {
-        using AddOutcome = RCLValidations::AddOutcome;
+        if(j.debug())
```

Comment:
Since this is at debug level and the others are at warn level, I'd rather keep this one all the time in case you are grepping the log for a specific level.